src/main/java/org/apache/commons/lang3/ArrayUtils.java

```diff
@@ -3566,7 +3566,7 @@ public static boolean isSameType(final Object array1, final Object array2) {
         if (array1 == null || array2 == null) {
             throw new IllegalArgumentException("The Array must not be null");
         }
-        return array1.getClass().getName().equals(array2.getClass().getName());
+        return ClassUtils.getName(array1).equals(ClassUtils.getName(array2));
```

Comment:
Why? See my comments in your other PR. In this case, neither array1 or array2 can be null here.

Reply:
Yep, my bad. changed.